std/range/package.d

```diff
@@ -6562,17 +6556,23 @@ if (isForwardRange!RangeOfRanges &&
             if (!e.empty)
             {
                 e.popFront();
-                r.front = e;
+                static if (hasAssignableElements!(ElementType!RangeOfRanges))
+                {
+                    r.front = e;
```

Comment:
Clarification requested: What is this supposed to be doing? It's assigning `r.front`, then immediately calls `r.popFront();`. IOW, it's only heating the CPU. 

Note that I'm not asking why I wrote the code I did, but why this line exists in the previous incarnation of this code.